src/libstd/sys/unix/os.rs

```diff
@@ -429,6 +430,11 @@ pub unsafe fn pipe() -> IoResult<(FileDesc, FileDesc)> {
     }
 }
 
+// FIXME: remove the error shim once old_io is gone
+pub unsafe fn pipe_shim() -> io::Result<(FileDesc, FileDesc)> {
+    pipe().map_err(|_| io::Error::last_os_error())
```

Comment:
I think this may not actually work out because `IoError::last_error` may allocate and could change errno.